src/basho_bench.erl

```diff
@@ -70,7 +72,10 @@ main(Args) ->
     application:set_env(lager, crash_log, CrashLog),
     lager:start(),
 
-    application:ensure_all_started(couch),
+    {ok, _} = application:ensure_all_started(couch),
+
+    ok = basho_bench_app:start(),
+    basho_bench_config:set(test_id, BenchName),
```

Comment:
I still think we're doing too much in the main function, or perhaps we need an alternative function for starting things from outside an escript. It's not obvious to me how you expect to run this, are you envisioning calling `basho_bench:main/1` and supplying params there? Seems awkward to go that route given all the extra logic main does around code paths and the other various bits.